http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/AppUsageComponent.scala

```diff
@@ -51,22 +54,33 @@ object appUsageQuery extends TableQuery(new AppUsageTable(_)) {
     } yield id
 
     for {
-      res <- dbio.transaction.attempt
+      res <- dbio.transaction(TransactionIsolation.Serializable).attempt
       id <- res match {
         case Left(e) if e.getMessage.contains("usage startTime was recorded previously") =>
           // We should alert if this happens
           metrics.incrementCounter("appStartUsageTimeRecordingFailure") >> logger.error(e)(e.getMessage) >> F
             .raiseError(e)
-        case Left(e)      => F.raiseError(e)
-        case Right(appId) => F.pure(appId)
+        case Left(e: com.mysql.cj.jdbc.exceptions.MySQLTransactionRollbackException)
+            if e.getMessage contains "Deadlock" =>
```

Comment:
Deadlock happens sometimes as revealed by the unit test I added...
```
{"@timestamp":"2024-03-08T20:09:01.787539Z","@version":"1","logger_name":"org.broadinstitute.dsde.workbench.leonardo.LeonardoTestSuite","thread_name":"io-compute-2","severity":"INFO","serviceContext":null,"message":"Deadlock detected, sleeping for 57 ms"}
```

This adds a bit jitter to recover from deadlock